component/topology/topology.go

```diff
@@ -38,8 +38,10 @@ func Subscribe() Subscriber {
 }
 
 func Stop() {
-	if syncer == nil {
-		return
+	if syncer != nil {
+		syncer.Stop()
+	}
+	if discover != nil {
+		_ = discover.Close()
 	}
```

Comment:
This is the only place I've found that behaves differently than before. Is this an existing leak problem? Does it have a big impact? Do we need cherry pick it individually?

Reply:
Only tests will be effected. That the discover keeps running during the lifetime of the program is expected.